yt_dlp/extractor/rcti.py

```diff
@@ -142,35 +191,44 @@ def _real_extract(self, url):
                 'id': 'landscape_image',
                 'url': '%s%d%s' % (image_path, 2000, video_meta['landscape_image'])
             })
-
-        formats = self._extract_m3u8_formats(video_url, display_id, 'mp4', headers={'Referer': 'https://www.rctiplus.com/'})
+        try:
+            formats = self._extract_m3u8_formats(video_url, display_id, 'mp4', headers={'Referer': 'https://www.rctiplus.com/'})
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
+                self.raise_geo_restricted(countries=['ID'], metadata_available=True)
+            raise e
         for f in formats:
-            if 'akamaized' in f['url']:
-                f.setdefault('http_headers', {})['Referer'] = 'https://www.rctiplus.com/'  # Referer header is required for akamai CDNs
+            if 'akamaized' in f['url'] or 'cloudfront' in f['url']:
+                f.setdefault('http_headers', {})['Referer'] = 'https://www.rctiplus.com/'  # Referer header is required for akamai/cloudfront CDNs
 
         self._sort_formats(formats)
 
         return {
             'id': video_meta.get('product_id') or video_json.get('product_id'),
-            'title': video_meta.get('title') or video_json.get('content_name'),
+            'title': (video_meta.get('title') or video_meta.get('name')
+                      or video_json.get('content_name') or video_json.get('assets_name')),
             'display_id': display_id,
             'description': video_meta.get('summary'),
-            'timestamp': video_meta.get('release_date'),
+            'timestamp': video_meta.get('release_date') or video_json.get('start_date'),
             'duration': video_meta.get('duration'),
-            'categories': [video_meta.get('genre')],
+            'categories': [video_meta['genre']] if video_meta.get('genre') else None,
             'average_rating': video_meta.get('star_rating'),
             'series': video_meta.get('program_title') or video_json.get('program_title'),
             'season_number': video_meta.get('season'),
             'episode_number': video_meta.get('episode'),
             'channel': video_json.get('tv_name'),
             'channel_id': video_json.get('tv_id'),
             'formats': formats,
-            'thumbnails': thumbnails
+            'thumbnails': thumbnails,
+            'is_live': video_type == 'live-event' and not is_upcoming,
+            'was_live': video_type == 'missed-event',
+            'live_status': 'is_upcoming' if is_upcoming else None,
```

Comment:
You can also just set `live_status` and the other two will be automatically set

Reply:
sure, but it honestly seems simpler this way